File: 프로그래머스/level3/2Xn타일링.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<int> tile;
    tile.push_back(0);
    tile.push_back(1);
    tile.push_back(2);
    for(int i=3; i<=n; i++){
        tile.push_back((tile[i-1]+tile[i-2])%1000000007);
    }
    answer = tile[n];
    return answer;
}

File: 프로그래머스/level3/2xn타일링.js
function solution(n) {
    var answer = 0;
    const dp = [0, 1, 2];
    for(let i=3; i<=n; i++){
        dp[i] = ((dp[i-2]+dp[i-1])%1000000007);
    }
    return dp[n];
}